rotatingtext/src/main/java/com/sdsmdg/harjot/rotatingtext/RotatingTextSwitcher.java

```diff
@@ -174,16 +174,31 @@ protected void onDraw(Canvas canvas) {
     }
 
     private void animateInHorizontal() {
-        ValueAnimator animator = ValueAnimator.ofFloat(0.0f, getHeight());
-        animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
-            @Override
-            public void onAnimationUpdate(ValueAnimator valueAnimator) {
-                pathIn = new Path();
-                pathIn.moveTo(0.0f, (Float) valueAnimator.getAnimatedValue() - paint.getFontMetrics().bottom);
-                pathIn.lineTo(getWidth(), (Float) valueAnimator.getAnimatedValue() - paint.getFontMetrics().bottom);
-                rotatable.setPathIn(pathIn);
-            }
-        });
+        ValueAnimator animator;
+        if(!rotatable.getApplyHorizontal()) {
+            animator = ValueAnimator.ofFloat(0.0f, getHeight());
+            animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+                @Override
+                public void onAnimationUpdate(ValueAnimator valueAnimator) {
+                    pathIn = new Path();
+                    pathIn.moveTo(0.0f, (Float) valueAnimator.getAnimatedValue() - paint.getFontMetrics().bottom);
+                    pathIn.lineTo(getWidth(), (Float) valueAnimator.getAnimatedValue() - paint.getFontMetrics().bottom);
+                    rotatable.setPathIn(pathIn);
+                }
+            });
+        }
+        else {
+            animator = ValueAnimator.ofFloat(-getWidth(),0.0f );
+            animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+                @Override
+                public void onAnimationUpdate(ValueAnimator valueAnimator) {
+                    pathIn = new Path();
+                    pathIn.moveTo((Float) valueAnimator.getAnimatedValue(), 2*getHeight()/3.0f);
+                    pathIn.lineTo((Float) valueAnimator.getAnimatedValue() +90000.0F, 2*getHeight()/3.0f);
```

Comment:
Instead of hard-coding a finishing point by adding 90,000....you can add getWidth() to determine the finishing point of the animation. This way there will be less load on the customView. Try this and tell me if there are any bugs on using this, otherwise it is working fine.